core/controllers/resources.py

```diff
@@ -115,6 +115,24 @@ def get(self, page_context, page_identifier, asset_type, encoded_filename):
 class PromoBarHandler(base.BaseHandler):
     """Handler for the promo-bar."""
 
+    URL_PATH_ARGS_SCHEMAS = {}
+    HANDLER_ARGS_SCHEMAS = {
+        'GET': {},
+        'PUT': {
+            'promo_bar_enabled': {
+                'schema': {
+                    'type': 'bool'
+                },
+                'default_value': False
```

Comment:
Why is this `False`?

Reply:
I was thinking of making it "None" instead of "False" but I was actually confused because it can't be none as It can either be true or false (i.e either enabled or either disabled). But still, if I have misunderstood something then kindly correct me.

PS: I had asked the meaning of false and none from Nikhil (@Nik-09) and he explained it as follows:
-----------------
default_value -- None: Means the argument is optional i.e., if it is absent then it won't create any error.
default_value -- False: Means if it is absent then it should be updated as false and other computation will be performed based on it.
-----------------
So from this explanation, I thought "False" would be more appropriate over here.

Reply:
But we also don't need to set it at all, right? You might need to check the frontend code whether we send the `promo_bar_enabled` every time.

Reply:
@vojtechjelinek  This is what I have got, please correct me if I am wrong somewhere!

[https://github.com/ayushjaink8/oppia/blob/develop/core/domain/config_domain.py](url)

**File:** core/domain/config_domain.py
![image](https://user-images.githubusercontent.com/75239620/140979506-472c9ec4-e467-4be8-b7f0-591116e197f7.png)
The default value is triggered from **line no: 369**  I guess as I tried changing the value from **False** to **True** and then I found out that the txt file which gets downloaded from the website also contains "true" instead of "false" as shown below:
![image](https://user-images.githubusercontent.com/75239620/140980728-36e773fb-da19-4163-85d1-626d0f0e82bb.png)

I have searched this class in the front end templates and got this:
PromoBar Class: [https://github.com/ayushjaink8/oppia/blob/develop/core/templates/domain/promo_bar/promo-bar.model.ts](url)
**File:** core/templates/domain/promo_bar/promo-bar.model.ts
![image](https://user-images.githubusercontent.com/75239620/140980164-4e49c7dd-4850-401f-ba88-7bd8962a21fb.png)

These are the Unit Tests for this particular class to make sure everything works fine:
**File:** [https://github.com/ayushjaink8/oppia/blob/develop/core/templates/domain/promo_bar/promo-bar.model.spec.ts](url)

**We are sending this every time because even if I remove this default value, it doesn't make a difference so I am removing this default value and updating my PR.**

If there is something which I missed here then please let me know.

Reply:
@vojtechjelinek 
I committed the requested changes
Check this commit: [https://github.com/oppia/oppia/pull/14171/commits/39ebc540f1863cea99f4e502e6bef7f0061e33e0](url)

Reply:
Looks good!